op-challenger/fault/game_test.go

```diff
@@ -0,0 +1,121 @@
+package fault
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+func createTestClaims() []Claim {
+	top := Claim{
+		Value:         common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000768"),
+		Position:      NewPosition(0, 0),
+		Parent:        nil,
+		DefendsParent: false,
+	}
+
+	middle := Claim{
+		Value:         common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000768"),
+		Position:      NewPosition(1, 1),
+		Parent:        &top,
+		DefendsParent: true,
+	}
+
+	bottom := Claim{
+		Value:         common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000465"),
+		Position:      NewPosition(2, 2),
+		Parent:        &middle,
+		DefendsParent: false,
+	}
+
+	return []Claim{top, middle, bottom}
+}
+
+// TestGame_Put_AlreadyExists tests the [Game.Put] method using a [gameState] instance
+// errors when a claim already exists.
+func TestGame_Put_AlreadyExists(t *testing.T) {
+	// Create a new game state.
+	g := NewGameState()
+
+	// Create a new claim.
+	claim := Claim{
+		Value:    common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000768"),
+		Position: NewPosition(1, 1),
+	}
+
+	// Put the claim into the game state.
+	err := g.Put(claim)
+	require.NoError(t, err)
+
+	// Put the claim into the game state again.
+	err = g.Put(claim)
+	require.ErrorIs(t, err, ErrClaimExists)
+}
+
+// TestGame_Put_ParentsAndChildren tests the [Game.Put] method using a [gameState] instance.
+func TestGame_Put_ParentsAndChildren(t *testing.T) {
+	// Create a new game state.
+	g := NewGameState()
+
+	// Create test claims
+	claims := createTestClaims()
+	top := claims[0]
+	middle := claims[1]
+	bottom := claims[2]
+
+	// Put the middle claim into the game state.
+	// We should expect no parent to exist, yet.
```

Comment:
This doesn't make sense - we should always be putting in the parent first.

Reply:
If we don't allow this, then there's this strange implicit guarantee that the game state will be constructed walking down from the root.

If that's the way we want to go, we should make this guarantee explicit and then require that the game state be constructed that way. This would probably change the api/interface of the game state though with the agent (especially it's construction).

Reply:
On the agent side I doubt it will change much. I guess it wasn't explicit, but I am fine saying that parents must be added before their children. If we iterate the state in the L1 contract we have this property